sched/scheduler/stateful_scheduler.go

```diff
@@ -92,11 +105,11 @@ func NewStatefulScheduler(
 
 	// TODO: we need to allow the scheduler to accept new jobs
 	// while recovering old ones.
-	if recoverJobsOnStartup {
+	if config.RecoverJobsOnStartup {
 		sched.startUp()
 	}
 
-	if !debugMode {
+	if !config.DebugMode {
```

Comment:
It really doesn't make any progress on DX-1645 other than letting us specify via config, instead of changing code.  

The goal for this review was to make adding config a non breaking and easier change you just add more variables to the config object.